src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -258,6 +258,23 @@ internal WatermarkOffsets GetWatermarkOffsets(string topic, int partition)
             return new WatermarkOffsets(low, high);
         }
 
+        internal IEnumerable<TopicPartitionOffset> OffsetsForTimes(IEnumerable<TopicPartitionTimestamp> timestampsToSearch, int millisecondsTimeout)
+        {
+            IEnumerable<TopicPartitionOffset> offsets = timestampsToSearch
+                .Select(t => new TopicPartitionOffset(t.TopicPartition, t.Timestamp.UnixTimestampMs))
+                .ToList();
+            IntPtr cOffsets = GetCTopicPartitionList(offsets);
+            ErrorCode err = LibRdKafka.offsets_for_times(handle, cOffsets, (IntPtr)millisecondsTimeout);
+            if (err != ErrorCode.NoError)
+            {
+                throw new KafkaException(err);
+            }
+
+            var list = GetTopicPartitionOffsetErrorList(cOffsets);
+            LibRdKafka.topic_partition_list_destroy(cOffsets);
+            return list.Select(t => t.TopicPartitionOffset).ToList();
```

Comment:
Not sure that I can repro the partition error when querying non-existent partition. In my case I've got a general error. 

But I've made checks for errors for every partition, and in case of any error `OffsetsForTimes` throws the `KafkaException` for the first found error.

Thoughts?